scripts/h5toms.py

```diff
@@ -335,83 +374,123 @@
-                    #    turns = np.outer((h5.w[:, cp_index] / katpoint.lightspeed) - cable_delay, h5.channel_freqs)
-                    #    vis_data *= np.exp(-2j * np.pi * turns)
-
-                    out_utc = utc_seconds
-                    out_freqs = h5.channel_freqs
-
-                    # Overwrite the input visibilities with averaged visibilities,flags,weights,timestamps,channel freqs
-                    if average_data:
-                        vis_data, weight_data, flag_data, out_utc, out_freqs = \
-                            averager.average_visibilities(vis_data, weight_data, flag_data, out_utc, out_freqs,
-                                                          timeav=dump_av, chanav=chan_av, flagav=options.flagav)
-
-                    pol_data.append(vis_data)
-                    weight_pol_data.append(weight_data)
-                    flag_pol_data.append(flag_data)
-
-                vis_data = np.dstack(pol_data)
-                weight_data = np.dstack(weight_pol_data)
-                flag_data = np.dstack(flag_pol_data)
-
-                model_data = None
-                corrected_data = None
-                if options.model_data:
-                    # unity intensity zero phase model data set, same shape as vis_data
-                    model_data = np.ones(vis_data.shape, dtype=np.complex64)
-                    # corrected data set copied from vis_data
-                    corrected_data = vis_data
-
-                uvw_coordinates = np.array(target.uvw(ant2, timestamp=out_utc, antenna=ant1))
-
-                # Convert averaged UTC timestamps to MJD seconds.
-                out_mjd = [katpoint.Timestamp(time_utc).to_mjd() * 24 * 60 * 60 for time_utc in out_utc]
-
-                # Increment the filesize.
-                sz_mb += vis_data.dtype.itemsize * vis_data.size / (1024.0 * 1024.0)
-                sz_mb += weight_data.dtype.itemsize * weight_data.size / (1024.0 * 1024.0)
-                sz_mb += flag_data.dtype.itemsize * flag_data.size / (1024.0 * 1024.0)
-
-                if options.model_data:
-                    sz_mb += model_data.dtype.itemsize * model_data.size / (1024.0 * 1024.0)
-                    sz_mb += corrected_data.dtype.itemsize * corrected_data.size / (1024.0 * 1024.0)
-
-                # write the data to the ms.
-                main_dict = ms_extra.populate_main_dict(uvw_coordinates, vis_data, flag_data,
-                                                        out_mjd, ant1_index, ant2_index,
-                                                        dump_time_width, field_id, state_id,
-                                                        scan_itr, model_data, corrected_data)
-                ms_extra.write_rows(main_table, main_dict, verbose=options.verbose)
+        # Iterate over time in some multiple of dump average
+        ntime = utc_seconds.size
+        tsize = dump_av
+        ntime_av = ntime/tsize
```

Comment:
`ntime_av` is only there to calculate the number of averaged timestamps for logging some output as h5toms runs. Previously it was calculated from the shape of the averaged timestamps directly, but now as the averager no longer sees all of the timestamps for a scan at once I compute it from the averaging length and the scan length. No output gets printed if `ntime_av==0` so it makes no difference in that case.
It's probably best that I move the assignment of `ntime_av` to just before the output gets printed so as to avoid confusion.